ios/FluentUI/Card Nudge/CardNudgeView.swift

```diff
@@ -0,0 +1,195 @@
+                            .font(.subheadline)
+                            .layoutPriority(1)
+                            .lineLimit(1)
+                            .foregroundColor(Color(tokens.accentColor))
+                    }
+                    if let subtitle = state.subtitle {
+                        Text(subtitle)
+                            .font(.subheadline)
+                            .lineLimit(1)
+                            .foregroundColor(Color(tokens.subtitleTextColor))
+                    }
+                }
+            }
+        }
+    }
+
+    @ViewBuilder
+    var buttons: some View {
+        HStack(spacing: 0) {
+            if let actionTitle = state.actionButtonTitle,
+                      let action = state.actionButtonAction {
+                SwiftUI.Button(actionTitle) {
+                    action(state)
+                }
+                .lineLimit(1)
+                .padding(.horizontal, tokens.buttonInnerPaddingHorizontal)
+                .padding(.vertical, tokens.buttonInnerPaddingVertical)
+                .foregroundColor(Color(tokens.accentColor))
+                .background(
+                    RoundedRectangle(cornerRadius: .infinity)
+                        .foregroundColor(Color(tokens.buttonBackgroundColor))
+                )
+            }
+            if let dismissAction = state.dismissButtonAction {
+                SwiftUI.Button(action: {
+                    dismissAction(state)
+                }, label: {
+                    Image("dismiss-20x20", bundle: FluentUIFramework.resourceBundle)
+                })
+                .padding(.horizontal, tokens.buttonInnerPaddingHorizontal)
+                .padding(.vertical, tokens.buttonInnerPaddingVertical)
+                .accessibility(identifier: "Accessibility.Dismiss.Label")
+                .foregroundColor(Color(tokens.textColor))
+            }
+        }
+    }
+
+    @ViewBuilder
+    var innerContents: some View {
+        HStack(spacing: 0) {
+            icon
+            textContainer
+            Spacer(minLength: 0)
+            buttons
+                .layoutPriority(1)
+        }
+        .padding(.vertical, tokens.mainContentVerticalPadding)
+        .padding(.horizontal, tokens.innerPadding)
+        .frame(minHeight: tokens.minimumHeight)
+    }
+
+    public var body: some View {
+        innerContents
+            .background(
+                RoundedRectangle(cornerRadius: tokens.cornerRadius)
+                    .strokeBorder(Color(tokens.backgroundBorderColor), lineWidth: 1.0)
+                    .background(
+                        RoundedRectangle(cornerRadius: tokens.cornerRadius)
+                            .fill(Color(tokens.backgroundColor))
+                    )
+            )
+            .padding(.vertical, tokens.outerVerticalPadding)
+            .padding(.horizontal, tokens.outerHorizontalPadding)
+    }
+
+    init(_ state: CardNudgeViewState) {
```

Comment:
I think it's better to create the state internally if we don't have this control being used in a composite control scenario.
Then we can provide modifiers for the SwiftUI consumption like we're prototyping for the Avatar.

Reply:
Yeah I'm a big fan of that approach now too; you've fully convinced me that the state object should be buried as deep as possible :) I'll knock that out ASAP!